lib/releases.js

```diff
@@ -80,74 +79,84 @@ const contributorsSentence = ({ commits, pullRequests }) => {
 }
 
 const getCategoriesConfig = ({ config }) => {
-  const categoriesConfig = config.categories.reduce((acc, category) => {
-    const labels = category.label ? [category.label] : category.labels
-    labels.forEach(label => (acc[label] = category))
-    return acc
-  }, {})
-
-  const orderedCategories = [
-    UNCATEGORIZED,
-    ..._.flatMap(config.categories, category =>
-      category.label ? category.label : category.labels
-    )
-  ]
-
-  return [categoriesConfig, orderedCategories]
+  const categoriesConfig = [{ title: UNCATEGORIZED }, ...config.categories]
+
+  return categoriesConfig.map(category => {
+    if (!category.labels) category.labels = []
+    if (category.label) {
+      // treat label as labels
+      category.labels.push(category.label)
+      delete category.label
+    }
+    return { ...category, pullRequests: [] }
+  })
 }
 
-const categorizePullRequests = (pullRequests, categories, excludeLabels) => {
-  return pullRequests.reduce(
-    (acc, pullRequest) => {
+const categorizePullRequests = (
+  pullRequests,
+  orderedCategories,
+  excludeLabels
+) => {
+  const categories = orderedCategories.flatMap(category => category.labels)
+
+  // exclude pull requests with exclude labels first than push any uncategorized pull requests to UNCATEGORIZED category
+  const filterPullRequests = pullRequest => {
+    const labels = pullRequest.labels.nodes
+
+    if (labels.some(label => excludeLabels.includes(label.name))) {
+      return false
+    } else if (
+      labels.length === 0 ||
+      !labels.some(label => categories.includes(label.name))
+    ) {
+      orderedCategories[0].pullRequests.push(pullRequest)
+      return false
+    }
+    return true
+  }
+
+  const filterUncategorized = category => category.title !== UNCATEGORIZED
+
+  // we only want pull requests that have yet to be categorized
+  const uncategorizedPullRequests = pullRequests.filter(filterPullRequests)
+
+  orderedCategories.filter(filterUncategorized).map(category => {
+    uncategorizedPullRequests.map(pullRequest => {
+      // lets categorize some pull request based on labels
+      // due note that having the same label in multiple categories
+      // then it is intended to "duplicate" the pull request into each cateogory
       const labels = pullRequest.labels.nodes
-      if (labels.some(label => excludeLabels.includes(label.name))) {
-        return acc
-      } else if (
-        labels.length === 0 ||
-        !labels.some(label => categories.includes(label.name))
-      ) {
-        acc[UNCATEGORIZED].push(pullRequest)
-      } else {
-        labels.forEach(label => {
-          if (!acc[label.name]) acc[label.name] = []
-
-          acc[label.name].push(pullRequest)
-        })
+      if (labels.some(label => category.labels.includes(label.name))) {
+        category.pullRequests.push(pullRequest)
```

Comment:
Also here we shouldn't mutate the input :)
We can create a new array which contains the results.

Reply:
Created a new array.